HDSAviz/Interactive_vis_toolkit.ipynb

```diff
@@ -0,0 +1,443 @@
+    "    \"\"\"\n",
+    "    path = \"c:/HDSAviz_data\"\n",
+    "    os.chdir(path)\n",
+    "    print 'current working directory: ' + os.getcwd()\n",
+    "\n",
+    "\n",
+    "def gather_all_analysis_files():\n",
+    "    \"\"\"\n",
+    "    This function gathers all the analysis files from C:/HDSAvis_data/ \n",
+    "    \n",
+    "    Output: \n",
+    "    Returns a list of analysis files that can be fed into plotting function\n",
+    "    \n",
+    "    \"\"\"\n",
+    "    new = !ls\n",
+    "    analysis_files = []\n",
+    "    for files in new:\n",
+    "        if files.startswith(\"analysis\"):\n",
+    "            analysis_files.append(files)\n",
+    "    print 'all analysis files gathered from current directory!' \n",
+    "    return analysis_files\n",
+    "\n",
+    "def plot_all_outcomes_burtin(S1_Range,ST_Range): \n",
+    "    \"\"\"\n",
+    "    This function plots the burtin type plots for all the outcome \n",
+    "    variables in different text files.\n",
+    "    \n",
+    "    Input :\n",
+    "    S1_Range: Minimum value of sensitivity for first order effects * 0.001, \n",
+    "    ST_Range: Minimum value of sensitivity for total order effects * 0.001\n",
+    "    \n",
+    "    Output:\n",
+    "    Tab based plotting of Graphs. \n",
+    "    \"\"\"\n",
+    "    working_dir = !pwd\n",
+    "    if working_dir != 'C:/HDSAviz_data/' :\n",
+    "        change_directory()\n",
+    "    \n",
+    "    analysis_files = gather_all_analysis_files()\n",
+    "    tabs_dictionary = {}\n",
+    "    \n",
+    "    for i,files in zip(range(len(analysis_files)), analysis_files):\n",
+    "        # cleaning the file names to get a short name to occupy less \n",
+    "        # space in tabs.\n",
+    "        names = files.replace('analysis_', '' ).replace('.txt','')\n",
+    "        \n",
+    "        try :\n",
+    "            tabs_dictionary[i] = Panel(child = makeplot(files, \n",
+    "                                np.array(['S1', 'ST']), \n",
+    "                                np.array([0.001*S1_Range, 0.001*ST_Range])),\n",
+    "                                title = names)    \n",
+    "        except:\n",
+    "            pass\n",
+    "\n",
+    "    \n",
+    "    tabs = Tabs(tabs=tabs_dictionary.values())\n",
+    "    \n",
+    "    try :\n",
+    "        p = show(tabs)\n",
+    "        output_notebook()\n",
+    "    except:\n",
+    "        pass\n",
+    "    \n",
+    "    return p\n",
+    "\n",
+    "def Interact_with_burtin_plots():\n",
+    "    \"\"\"\n",
+    "    This function adds capability to adjust the minimum threshold for \n",
+    "    sensitiity plotting\n",
+    "    \n",
+    "    Output: \n",
+    "    Interactive graph displaying a slider widgets that allow for \n",
+    "    minimum threshold of sensitivity selection\n",
+    "    \"\"\"\n",
+    "    output_notebook()\n",
+    "    slider1 = FloatSlider(value = 1, min = 0.1, max = 10, \n",
```

Comment:
If so, we should look at the impact on interactive performance.  If the slider step size is 0.001 then sliding from 0 to 0.5 will require the computer to compute new values 500 times before you get from where you are going to what you want to see...  This is the main problem with sliders in my opinion, they calculate everything in between your start and end points.